src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -138,50 +138,47 @@ public async Task FindAsync<TArg>(
                 onAssetFound(this.SourceGeneratorExecutionVersionMap, filteredExecutionMap, arg);
             }
 
-            if (compilationState.FrozenSourceGeneratedDocumentStates != null)
-            {
-                Contract.ThrowIfFalse(FrozenSourceGeneratedDocumentIdentities.HasValue);
-                Contract.ThrowIfFalse(FrozenSourceGeneratedDocuments.HasValue);
+            Contract.ThrowIfFalse(FrozenSourceGeneratedDocumentIdentities.HasValue);
+            Contract.ThrowIfFalse(FrozenSourceGeneratedDocuments.HasValue);
```

Comment:
If these should now always have a value, should they become non-nullable too?

Reply:
So much simpler.  Thanks!